packages/test/test-version-utils/src/compatConfig.ts

```diff
@@ -170,6 +173,47 @@ if (compatKind !== undefined) {
     configList = configList.filter((value) => compatKind!.includes(value.kind));
 }
 
+function getUnexpectedLogErrorException(logger: ErrorTrackingLogger){
+    const results = logger.results();
+    if(results.unexpectedErrors.length > 0){
+        return new Error(
+            // eslint-disable-next-line max-len
+            `Unexpected Errors in Logs. Use itExpects to specify expected errors:\n${ JSON.stringify(results.unexpectedErrors, undefined, 2)}`);
+    }
+    if(results.expectedNotFound.length > 0){
+        return new Error(
+            `Expected Events not found:\n${ JSON.stringify(results.expectedNotFound, undefined, 2)}`);
+    }
+}
+
+function createExpectsTest(expectedEvents: ITelemetryGenericEvent[], test: Mocha.AsyncFunc){
+    return async function (this:Context){
+        const provider: TestObjectProvider | undefined = this.__fluidTestProvider;
+        assert(provider !== undefined, "Expected __fluidTestProvider on this");
+        try{
+            provider.logger.registerExpectedEvent(... expectedEvents);
+            await test.bind(this)();
+        }catch(error){
+            provider.logger.sendErrorEvent({eventName:"TestException"},error)
+        }
+        const err = getUnexpectedLogErrorException(provider.logger);
+        if(err !== undefined){
+            throw err;
+        }
+    };
+}
+
+export type ExpectsTest = (name: string, expectedEvents: ITelemetryGenericEvent[], test: Mocha.AsyncFunc) => Mocha.Test
+
+export const itExpects: ExpectsTest & {only: ExpectsTest} =
```

Comment:
After looking at the usage, I think I would prefer if it read this way:

 ```
it("testName", async () => {
  /* test code */
}).expects([
  /* expected logs */
]);
```

I haven't tried it, and not sure how much trouble it is to mixin an `expects` function onto Mocha's `it`, but I can imagine it could be possible.  What do you think?

Reply:
i don't think it is possible in a way that keeps any typing

Reply:
Would this direction work?  Didn't try running it yet, or putting this code in the right place, but there aren't any red squiggles at least.  Anyway, obviously it's fine to proceed as-is, but I do like this style better if it works.

```
declare module "mocha" {
    class ExpectantTest extends Test {
        expects(foo: string): Test;
    }

    interface TestFunction {
        (title: string, fn?: Func): ExpectantTest;
        (title: string, fn?: AsyncFunc): ExpectantTest;
    }
}

it = Object.assign(it, { expects: (foo: string) => { /* whatever */ } });

it("some test", () => {
    // whatever
}).expects("stuff");
```